backend/app/providers/telegram.go

```diff
@@ -0,0 +1,65 @@
+package providers
+
+// This provider provides common logic, reused by Telegram auth and notifications verification
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+
+	"github.com/umputun/remark42/backend/app/notify"
+)
+
+type tgRequester interface {
+	Request(ctx context.Context, method string, b []byte, data interface{}) error
+}
+
+// TGUpdatesReceiver used to dispatch telegram updates to multiple receivers
+type TGUpdatesReceiver interface {
+	ProcessUpdate(ctx context.Context, textUpdate string) error
+}
+
+// DispatchTelegramUpdates dispatches telegram updates to provided list of receivers
+// Blocks caller
+func DispatchTelegramUpdates(ctx context.Context, requester tgRequester, receivers []TGUpdatesReceiver, period time.Duration) {
+	var updateOffset int
+
+	processUpdatedTicker := time.NewTicker(period)
+	for {
+		select {
+		case <-ctx.Done():
+			processUpdatedTicker.Stop()
+			return
+		case <-processUpdatedTicker.C:
+			url := `getUpdates?allowed_updates=["message"]`
+			if updateOffset != 0 {
+				url += fmt.Sprintf("&offset=%d", updateOffset)
+			}
+
+			var result notify.TelegramUpdate
+
+			err := requester.Request(ctx, url, nil, &result)
+			if err != nil {
+				log.Printf("[ERROR] failed to fetch updates: %v", err)
```

Comment:
I think this should be WARN and not ERROR, as ERROR usually means - the usual flow altered. Here you noe even doing continue (maybe missed?)